### PR TITLE
added test for InlineValidator

### PR DESCRIPTION
**Pull request by ahmetakbn**
### Subject

Test to increase code coverage of `SonataCoreBundle`

### CODE REVIEW

File: Tests/Validator/InlineValidatorTest.php [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        $this->context = $this->getMock(
+            interface_exists('Symfony\Component\Validator\Context\ExecutionContextInterface') ?
+                'Symfony\Component\Validator\Context\ExecutionContextInterface' :
+                'Symfony\Component\Validator\ExecutionContextInterface'
+        );
+    }
+
+    public function testGetErrorElement()
+    {
+        $inlineValidator = new InlineValidator($this->container, $this->constraintValidatorFactory);
```
Comment (greg0ire) [inline Tests/Validator/InlineValidatorTest.php]: Mother of god… do all classes depend on the container or what? T___T

File: Tests/Validator/InlineValidatorTest.php [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        $reflectedMethod->setAccessible(true);
+
+        $errorElement = $reflectedMethod->invokeArgs($inlineValidator, array('foo'));
+
+        $this->assertInstanceOf('Sonata\CoreBundle\Validator\ErrorElement', $errorElement);
+        $this->assertSame('foo', $errorElement->getSubject());
+    }
+
+    /**
+     * @expectedException \Exception
```
Comment (OskarStark) [inline Tests/Validator/InlineValidatorTest.php]: please use `$this->setExpectedException` instead of the annoations

File: Tests/Validator/InlineValidatorTest.php (hunk 1 above)
Comment (OskarStark) [inline Tests/Validator/InlineValidatorTest.php]: > Mother of god… do all classes depend on the container or what? T___T

😆

Comment (greg0ire) [inline Tests/Validator/InlineValidatorTest.php]: I think this should have been done with tagged services or something like that…

File: Tests/Validator/InlineValidatorTest.php (hunk 2 above)
Comment (core23) [inline Tests/Validator/InlineValidatorTest.php]: Nope, `setExpectedException` is deprecated, since phpunit 5.2.8

Comment (OskarStark) [inline Tests/Validator/InlineValidatorTest.php]: really?

File: Tests/Validator/InlineValidatorTest.php (hunk 1 above)
Comment (OskarStark) [inline Tests/Validator/InlineValidatorTest.php]: but please lets do this in another PR

File: Tests/Validator/InlineValidatorTest.php (hunk 2 above)
Comment (OskarStark) [inline Tests/Validator/InlineValidatorTest.php]: so we keep it this way?

File: Tests/Validator/InlineValidatorTest.php [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    }
+
+    public function testGetErrorElement()
+    {
+        $inlineValidator = new InlineValidator($this->container, $this->constraintValidatorFactory);
+
+        $reflectorClass = new \ReflectionClass($inlineValidator);
+        $reflectedProperty = $reflectorClass->getProperty('context');
+        $reflectedProperty->setAccessible(true);
+        $reflectedProperty->setValue($inlineValidator, $this->context);
```
Comment (greg0ire) [inline Tests/Validator/InlineValidatorTest.php]: Why not use `initialize` here ?

File: Tests/Validator/InlineValidatorTest.php (hunk 2 above)
Comment (core23) [inline Tests/Validator/InlineValidatorTest.php]: You shouldn't expect the god `\Exception` class

File: Tests/Validator/InlineValidatorTest.php [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+            'Symfony\Component\Validator\ConstraintValidatorFactoryInterface'
+        );
+        $this->context = $this->getMock(
+            interface_exists('Symfony\Component\Validator\Context\ExecutionContextInterface') ?
+                'Symfony\Component\Validator\Context\ExecutionContextInterface' :
+                'Symfony\Component\Validator\ExecutionContextInterface'
+        );
+    }
+
+    public function testGetErrorElement()
```
Comment (greg0ire) [inline Tests/Validator/InlineValidatorTest.php]: Not a huge fan of testing protected methods… you should only test them trough the public API of the object IMO : we don't really care about the implementation, only about the behavior of the system under test. If it is not simple to test like though, it might be justified…

File: Tests/Validator/InlineValidatorTest.php (hunk 2 above)
Comment (Soullivaneuh) [inline Tests/Validator/InlineValidatorTest.php]: > so we keep it this way?

No. Please use this method.

The method is deprecated, not removed.

We will change it when we can drop PHPUnit 4 usage.

Comment (greg0ire) [inline Tests/Validator/InlineValidatorTest.php]: Related : https://github.com/sebastianbergmann/phpunit/issues/2074

File: Tests/Fixtures/Bundle/Validator/FooValidatorService.php [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\CoreBundle\Tests\Fixtures\Bundle\Validator;
+
+use Sonata\CoreBundle\Validator\ErrorElement;
+
+/**
+ * Class FooValidatorService.
```
Comment (Soullivaneuh) [inline Tests/Fixtures/Bundle/Validator/FooValidatorService.php]: Useless comment.

File: Tests/Validator/InlineValidatorTest.php (hunk 2 above)
Comment (greg0ire) [inline Tests/Validator/InlineValidatorTest.php]: It's a bit tricky : it is deprecated, but not in favor of annotations. It's deprecated in favor of several methods that are not available to us : `expectException`, etc…

File: Tests/Validator/InlineValidatorTest.php (hunk 1 above)
Comment (greg0ire) [inline Tests/Validator/InlineValidatorTest.php]: > but please lets do this in another PR

Sure goes without saying, this is more a call to action…

File: Tests/Validator/InlineValidatorTest.php (hunk 2 above)
Comment (Soullivaneuh) [inline Tests/Validator/InlineValidatorTest.php]: > It's a bit tricky : it is deprecated, but not in favor of annotations. It's deprecated in favor of several methods that are not available to us : `expectException`, etc…

So just use methods which are both available on PHPUnit 4 and 5. :-)

Comment (ahmetakbn) [inline Tests/Validator/InlineValidatorTest.php]: So I will keep using annotations?

Comment (greg0ire) [inline Tests/Validator/InlineValidatorTest.php]: Uuuuh… nope!

File: Tests/Validator/InlineValidatorTest.php (hunk 3 above)
Comment (ahmetakbn) [inline Tests/Validator/InlineValidatorTest.php]: :dizzy_face: ah
Yes of course. I didn't really analyze the parent class, just didn't see any setter for the property.

File: Tests/Validator/InlineValidatorTest.php (hunk 4 above)
Comment (ahmetakbn) [inline Tests/Validator/InlineValidatorTest.php]: First of all, just trying to write test for every line which is possible. Form my point of view the protected methods are not %100 encapsulated, so testing them when it is possible/easy is always beneficial.

File: Tests/Fixtures/Bundle/Validator/FooValidatorService.php (hunk 5 above)
Comment (ahmetakbn) [inline Tests/Fixtures/Bundle/Validator/FooValidatorService.php]: This has been created automaticaly by the IDE. I will change it ok.

File: Tests/Validator/InlineValidatorTest.php (hunk 4 above)
Comment (greg0ire) [inline Tests/Validator/InlineValidatorTest.php]: Do as you see fit, just making sure you know my point of view. If it is really a hassle to test the protected method through the public methods, then by all means, do the reflection dark magic (just my opinion though, not sure what @sonata-project/contributors think about this).

File: Tests/Fixtures/Bundle/Validator/FooValidatorService.php (hunk 5 above)
Comment (Soullivaneuh) [inline Tests/Fixtures/Bundle/Validator/FooValidatorService.php]: I know I think we have the same. :+1: 

You can change it on settings.

File: Tests/Validator/InlineValidatorTest.php (hunk 2 above)
Comment (ahmetakbn) [inline Tests/Validator/InlineValidatorTest.php]: I didn't really understand what I really should use finally :smile:

Comment (greg0ire) [inline Tests/Validator/InlineValidatorTest.php]: You're supposed to use `$this->setExpectedException('Some\PreciseException', 'some exception message')`

File: Tests/Validator/InlineValidatorTest.php (hunk 3 above)
Comment (ahmetakbn) [inline Tests/Validator/InlineValidatorTest.php]: done.

File: Tests/Validator/InlineValidatorTest.php (hunk 2 above)
Comment (ahmetakbn) [inline Tests/Validator/InlineValidatorTest.php]: done

File: Tests/Fixtures/Bundle/Validator/FooValidatorService.php (hunk 5 above)
Comment (ahmetakbn) [inline Tests/Fixtures/Bundle/Validator/FooValidatorService.php]: I have changed the comment.

File: Tests/Validator/InlineValidatorTest.php [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+        $this->assertInstanceOf('Sonata\CoreBundle\Validator\ErrorElement', $errorElement);
+        $this->assertSame('foo', $errorElement->getSubject());
+    }
+
+    public function testValidateWithConstraintIsClosure()
+    {
+        $this->setExpectedException('Symfony\Component\Validator\Exception\ValidatorException', 'foo is equal to foo');
+
+        $constraint = $this->getMock('Symfony\Component\Validator\Constraint', array('isClosure', 'getClosure'));
```
Comment (greg0ire) [inline Tests/Validator/InlineValidatorTest.php]: `getMock` is deprecated in favor of `createMock`

Comment (Soullivaneuh) [inline Tests/Validator/InlineValidatorTest.php]: Yes, but `createMock` is not available under PHPUnit 4. So let it as is for now.

Comment (greg0ire) [inline Tests/Validator/InlineValidatorTest.php]: Oh, right… nevermind, then.

**Comment by ahmetakbn** [comment]
This PR is ready from my side

File: Tests/Fixtures/Bundle/Validator/FooValidatorService.php [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+use Sonata\CoreBundle\Validator\ErrorElement;
+use Symfony\Component\Validator\Exception\ValidatorException;
+
+/**
+ * Validator service to create exception for test.
+ *
+ * @author Ahmet Akbana <ahmetakbana@gmail.com>
+ */
+class FooValidatorService
```
Comment (Soullivaneuh) [inline Tests/Fixtures/Bundle/Validator/FooValidatorService.php]: `final`

Comment (ahmetakbn) [inline Tests/Fixtures/Bundle/Validator/FooValidatorService.php]: done.

File: Tests/Validator/InlineValidatorTest.php [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+namespace Sonata\CoreBundle\Tests\Validator;
+
+use Sonata\CoreBundle\Tests\Fixtures\Bundle\Validator\FooValidatorService;
+use Sonata\CoreBundle\Validator\ErrorElement;
+use Sonata\CoreBundle\Validator\InlineValidator;
+use Symfony\Component\Validator\Exception\ValidatorException;
+
+/**
+ * Test for InlineValidator.
```
Comment (Soullivaneuh) [inline Tests/Validator/InlineValidatorTest.php]: For me this comment is useless.

By `class InlineValidatorTest`, we know this class is to test `InlineValidator`. :wink:

Comment (ahmetakbn) [inline Tests/Validator/InlineValidatorTest.php]: done

**Comment by Soullivaneuh** [comment]
RTM. Thank you @ahmetakbn! :+1:

**Comment by OskarStark** [comment]
Good job @ahmetakbn !
